src/qrcode/utils.ts

```diff
@@ -114,63 +120,78 @@ function* handleSecureSend(
   return true
 }
 
-export function* handleBarcode(
-  barcode: QrCode,
-  e164NumberToAddress: E164NumberToAddressType,
-  recipientInfo: RecipientInfo,
-  secureSendTxData?: TransactionDataInput,
-  requesterAddress?: string
-) {
-  const walletConnectEnabled: boolean = yield* call(isWalletConnectEnabled, barcode.data)
+// Catch all handler for QR Codes
+// includes support for WalletConnect, hooks, and send flow (non-secure send)
+export function* handleQRCodeDefault({ qrCode }: HandleQRCodeDetectedAction) {
+  const walletConnectEnabled: boolean = yield* call(isWalletConnectEnabled, qrCode.data)
+
   // Regex matches any 40 hexadecimal characters prefixed with "0x" (case insensitive)
-  if (/^0x[a-f0-9]{40}$/gi.test(barcode.data)) {
-    barcode.data = `celo://wallet/pay?address=${barcode.data}`
+  if (/^0x[a-f0-9]{40}$/gi.test(qrCode.data)) {
+    qrCode.data = `celo://wallet/pay?address=${qrCode.data}`
```

Comment:
if you moved this part down to just before `let qrData: UriData` (which is safe because the next if statements are mutually exclusive with this one) then it becomes the same as 11 lines of handleQRCodeSecureSend and you can create a shared helper for both